src/components/Card.js

```diff
@@ -12,32 +15,55 @@ export class Card {
     return this.cardElement;
   }
 
-  generateCard() {
+  generateCard(cardDataObj) {
     this.cardElement = this._getTemplate();
     this.cardElement.querySelector(".card__image").src = this._link;
     this.cardElement.querySelector(".card__image").alt = this._title;
     this.cardElement.querySelector(".card__title").textContent = this._title;
-    this._setEventListeners();
+    this._setEventListeners(cardDataObj);
     return this.cardElement;
   }
 
-  _setEventListeners() {
+  _setEventListeners(cardDataObj) {
+    this.checkLike(cardDataObj);
     this.cardElement
       .querySelector(".card__heart-button")
       .addEventListener("click", () => {
-        this.cardElement
-          .querySelector(".card__heart-button")
-          .classList.toggle("card__heart-button_active");
+        this._handleLikeButton(this, cardDataObj);
+        // console.log(this);
+        cardDataObj.isLiked = !cardDataObj.isLiked;
```

Comment:
`        cardDataObj.isLiked = !cardDataObj.isLiked;
`

Here you make `isLiked` **toggled**, but when you call `_handleLikeButton`, there can be an **error**, so you'll get `isLiked` changed (but you shouldn't)


So, put this line into `checkLike`